applications/FluidDynamicsApplication/python_scripts/fluid_dynamics_analysis.py

```diff
@@ -15,6 +15,15 @@ class FluidDynamicsAnalysis(AnalysisStage):
     def __init__(self,model,parameters):
         # Deprecation warnings
         solver_settings = parameters["solver_settings"]
+
+        if solver_settings.Has("domain_size") and parameters["problem_data"].Has("domain_size"):
+            warn_msg = '"domain_size" defined twice, using the one in "solver_settings"'
+            Kratos.Logger.PrintWarning("FluidDynamicsAnalysis", warn_msg)
+
+        if solver_settings.Has("model_part_name") and parameters["problem_data"].Has("model_part_name"):
```

Comment:
ouch, 
        what is the one we should use then? what if the solve has subsolvers, each with a domain_size defined inside? (got crazy exactly with this with the combined solver we are using).

i am not against this PR, however in the long run we need a better solution than this.
for example, i also think that an analysis should not know the settings of problem data, since we need pass just a block of the settings and not the entire parameters.json.

This last comment is very important in order to be able to go to a multi stage solution. For example say that we will want to do a static stage followed by en eigenvalue one

in that case our Parameters.json should look lile

    {
           "problem_data" : {},
           "stages" : [
               { ... here settings for static stage ... },
               { ... here settings for eigenvalue stage ... }
           ]
   }

Reply:
@RiccardoRossi I agree with you (I hope I understand correctly, otherwise please correct me)
These entire blocks [Fluid](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/FluidDynamicsApplication/python_scripts/fluid_dynamics_analysis.py#L17-L26) and [Structure](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_analysis.py#L35-L50) are ONLY there for backwards compatibility

I one defines it properly, i.e. as you describe, then these warnings would not be needed

=> This needs some changes in the Interface, but nothing seems to be happening for some weeks already even though I posted some issue already

Reply:
@philbucher so we agree about assuming that Stage does not know the problem_data block?

regarding the definition, one problem is that the "correct" way in a sense would be to define it once for all in the problem_data, and treat it as a sort of global variable. 

regarding changes in the interface, i guess that not much will be happening in August. For this reason i would definitely postpone any change to the settings to after the holidays (end of august or beginning of september).
Note that i do accept this specific PR (only see the proposal for changing the warning text), however i think we'll need to change the mechanism in the future